File: server_nodejs/src/models/item.js
module.exports = (sequelize, Sequelize) => {
    class Item extends Sequelize.Model {}

    Item.init(
        {
            uid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            categoryid: {
                type: Sequelize.UUID,
                allowNull: false,
            },
            brandid: {
                type: Sequelize.UUID,
                allowNull: false,
            },
            name: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            specification: {
                type: Sequelize.TEXT('long'),
                allowNull: false,
            },
            description: {
                type: Sequelize.TEXT('long'),
                allowNull: false,
            },
            price: {
                type: Sequelize.FLOAT,
                allowNull: false,
            },
            stock_qty: {
                type: Sequelize.INTEGER,
                allowNull: false,
            },
        },
        {
            sequelize,
            modelName: 'item',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    Item.associate = (models) => {
        Item.belongsTo(models.item_category, {
            foreignKey: 'categoryid',
        });
        Item.belongsTo(models.brand, {
            foreignKey: 'brandid',
        });
        Item.hasMany(models.cart_item, {
            foreignKey: 'itemid',
        });
        Item.hasMany(models.item_img, {
            foreignKey: 'itemid',
            as: 'item_img',
        });
        Item.hasMany(models.order_item, {
            foreignKey: 'itemid',
        });
    };
    return Item;
};

File: server_asp/src/models/Item.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class Item
{
    public Guid Uid { get; set; }

    public Guid Categoryid { get; set; }

    public Guid Brandid { get; set; }

    public string Name { get; set; } = null!;

    public string Specification { get; set; } = null!;

    public string Description { get; set; } = null!;

    public float Price { get; set; }

    public int StockQty { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Brand Brand { get; set; } = null!;

    public virtual ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();

    public virtual ItemCategory Category { get; set; } = null!;

    public virtual ICollection<ItemImg> ItemImgs { get; set; } = new List<ItemImg>();

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}
